get-started/nodejs/build-images.md

```diff
@@ -0,0 +1,224 @@
+```
+
+## Create a Dockerfile for Node.js
+
+A Dockerfile is a text document that contains all the commands a user could call on the command line to assemble an image. When we tell Docker to build our image by executing the `docker run` command, Docker reads these instructions and executes them one by one and creates a Docker image as a result.
+
+Let’s walk through the process of creating a Dockerfile for our application. In the root of your working directory, create a file named `Dockerfile` and open this file in your text editor.
+
+> **Note**
+>
+> The name of the Dockerfile is not important but the default filename for many commands is simply `Dockerfile`. So, we’ll use that as our filename throughout this series.
+
+The first thing we need to do is to add a line in our Dockerfile that tells Docker what base image we would like to use for our application.
+
+Dockerfile:
+`FROM node:12.18.1`
+
+Docker images can be inherited from other images. Therefore, instead of creating our own base image, we’ll use the official Node.js image that already has all the tools and packages that we need to run a Node.js application. You can think of this in the same way you would think about class inheritance in object oriented programming. For example, if we were able to create Docker images in JavaScript, we might write something like the following.
+
+`class MyImage extends NodeBaseImage {}`
+
+This would create a class called `MyImage` that inherited functionality from the base class `NodeBaseImage`.
+
+In the same way, when we use the `FROM` command, we tell Docker to include in our image all the functionality from the `node:12.18.1` image.
+
+>**Note**
+>
+> If you want to learn more about creating your own base images, see [Creating base images](https://docs.docker.com/develop/develop-images/baseimages/).
+
+To make things easier when running the rest of our commands, let’s create a working directory. This instructs Docker to use this path as the default location for all subsequent commands. This way we do not have to type out full file paths but can use relative paths based on the working directory.
+
+`WORKDIR /app`
+
+Usually the very first thing you do once you’ve downloaded a project written in Node.js is to install npm packages. This will ensure that your application has all its dependencies installed into the `node_modules` directory where the Node runtime will be able to find them.
+
+Before we can run `npm install`, we need to get our `package.json` and `package-lock.json` files into our images. We use the `COPY` command to do this. The  `COPY` command takes two parameters. The first parameter tells Docker what file(s) you would like to copy into the image. The second parameter tells Docker where you want that file(s) to be copied to. We’ll copy the `package.json` and `package-lock.json` file into our working directory `/app`.
+
+```
+COPY package.json package.json
+COPY package-lock.json package-lock.json
```

Comment:
The VSCode Docker extension uses this nice single `COPY` instruction to copy both `package.json` and `package-lock.json` in just one layer instead of two separate layers.

```dockerfile
COPY ["package.json", "package-lock.json*", "./"]
```